includes/Checker/Abstract_Check_Runner.php

```diff
@@ -486,6 +486,9 @@ final public function get_plugin_basename() {
 			if ( is_dir( $plugin ) ) {
 				$this->plugin_basename = $plugin;
 			} else {
+				if ( filter_var( $plugin, FILTER_VALIDATE_URL ) ) {
+					$plugin = Plugin_Request_Utility::download_plugin( $plugin );
```

Comment:
`$plugin` value could be a string or boolean. May be we should handle case of false value also.

Reply:
While testing I gave valid URL but file system could extract the zip so returned false in that case. Another case, even if zip is extracted, it could not move folder using move_dir().

Reply:
I have drafted a PR to use download_dir() and directly unzip file to the target destination. You can take reference from the PR. https://github.com/WordPress/plugin-check/pull/543/files

Reply:
I think that is better to have in one PR. I'm confused to what to do, merge or keep one?